eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -132,6 +133,8 @@ trait Eclair {
 
   def onChainTransactions(count: Int, skip: Int): Future[Iterable[WalletTransaction]]
 
+  def sendWithPreimage(externalId_opt: Option[String], recipientNodeId: PublicKey, amount: MilliSatoshi, maxAttempts_opt: Option[Int] = None, feeThresholdSat_opt: Option[Satoshi] = None, maxFeePct_opt: Option[Double] = None)(implicit timeout: Timeout): Future[UUID]
```

Comment:
nit: can you put it right below the `send` method? This trait is a bit of a mess, let's try to add new methods close to related ones (same for the implementation below).

Reply:
Done